### PR TITLE
Fix system caption style background being used when cue style is provided.

### PR DESCRIPTION
**Pull request by BetweenTwoBits**
This is a proposed fix for issue #4080 

- [x] Fix caption background issue
    - [x] applyEmbeddedStyles && cue styles available
        - Use cue styles
    - [x] applyEmbeddedStyles && cue styles unavailable && captioningManager.isEnabled
        - Use user style from CaptioningManager
    - [x] applyEmbeddedStyles && cue styles unavailable && !captioningManager.isEnabled
        - Use CaptionStyleCompat defaults
    - [x] !applyEmbeddedStyles && captioningManager.isEnabled
        - Use user style from CaptioningManager
    - [x] !applyEmbeddedStyles && !captioningManager.isEnabled
        - Use CaptionStyleCompat defaults
- [x] Ensure caption manager is enabled before using its settings

### CODE REVIEW

**Comment by BetweenTwoBits** [comment]
Re PRing https://github.com/google/ExoPlayer/pull/4085 with dev-v2 as the base.

**Comment by BetweenTwoBits** [comment]
@tonihei Hopefully this one is based on the right branch :)

**Comment by tonihei** [comment]
Yes, that's the right branch ;) 

The change looks good in general, just waiting for @ojw28 as he's more knowledgeable about the various issues around subtitles formats.

**Comment by BetweenTwoBits** [comment]
Checking in on this PR, any update @tonihei?

**Comment by ojw28** [comment]
This pull request doesn't appear to fix the ref'd issue. If I use the test content provided in the issue and follow the reproduction steps, I see that the issue is still present.

I'll reply on the issue with more detail about how I think the problem should be fixed. I'm not sure the approach proposed in this change is the best one, although I'm not 100% certain of this!

**Comment by BetweenTwoBits** [comment]
Oh goodness, I must have screwed up the most recent merge. I'll get it fixed!

File: library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitleView.java [hunk 1]
```diff
@@ -206,7 +206,7 @@ public void setApplyEmbeddedFontSizes(boolean applyEmbeddedFontSizes) {
    * {@link CaptioningManager#getUserStyle()}, or to a default style before API level 19.
    */
   public void setUserDefaultStyle() {
-    setStyle(Util.SDK_INT >= 19 && !isInEditMode()
+    setStyle(Util.SDK_INT >= 19 && isCaptionManagerEnabled() &&!isInEditMode()
```
Comment (ojw28) [inline library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitleView.java]: This looks like a good change; thanks!

File: library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitlePainter.java [hunk 2]
```diff
@@ -245,7 +248,14 @@ private void setupTextLayout() {
     if (applyEmbeddedFontSizes && applyEmbeddedStyles) {
       cueText = this.cueText;
     } else if (!applyEmbeddedStyles) {
-      cueText = this.cueText.toString(); // Equivalent to erasing all spans.
+      if (Color.alpha(backgroundColor) > 0) {
```
Comment (ojw28) [inline library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitlePainter.java]: As discussed, for this change I think we should just add the background wherever the cue hasn't provided one (which is approximately what happened before, with the exception of where the cue provided one that was semi-transparent). Let's discuss styling priorities etc separately. I'm fairly sure you can do this by adding the span at L262 and setting the flags to Spanned.SPAN_PRIORITY.

**Comment by ojw28** [comment]
It would be good to get this merged. If possible, could you squash the commits down into one or two logical changes (given the end result is small it's fine just to squash them all into one commit).

**Comment by ojw28** [comment]
By the way, you should be able to delete the `cornerRadius` member variable as part of this change :).

File: library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitlePainter.java (hunk 2 above)
Comment (BetweenTwoBits) [reply library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitlePainter.java]: Adjustments made.

**Comment by ojw28** [comment]
Looks good to merge; thanks! Would it be possible for you to squash all the commits together though?

**Comment by BetweenTwoBits** [comment]
I messed up the squash on this one and github was showing 0 changes. Re-opened as #4411

**Pull request closed without merging**